main.tf

```diff
@@ -115,6 +162,16 @@ resource "azuread_application_identifier_uri" "this" {
   identifier_uri = each.value
 }
 
+resource "azuread_application_pre_authorized" "this" {
+  for_each = { for client in var.pre_authorized_client_applications : client.client_id => client }
+
+  application_id       = azuread_application.this.id
+  authorized_client_id = each.value.client_id
+
+  # This is a workaround to point to the correct permission IDs as we genereate the random_uuids for each permission scope
+  permission_ids = [for permission in each.value.permissions : random_uuid.oauth2_permission_scope[permission].result]
```

Comment:
Is this something we actually want to do **always**, our would it make more sense to just output our randomly generated UUIDs out of the module and allow creation of `azuread_application_pre_authorized` resources outside of the module?

Reply:
In my mind they are so closely related that it makes sense to keep them together. It also reduces the complexity for the user when using for_each on the module. But i would not be against returning the UUIDs to make it possible to for the user to create these outside of the module as well.

Reply:
I have implemented the output-approach in #15, which is more in-line with how we usually implement these kinds of features in the Baseline modules.